swig/include/python/gdal_python.i

```diff
@@ -5171,6 +5171,120 @@ def quiet_warnings():
     finally:
         PopErrorHandler()
 
+
+def algorithm(path):
```

Comment:
Is it confusing to have both `gdal.algorithm` and `gdal.Algorithm` ? Should `algorithm` become `Algorithm.__init__` ?

Reply:
good point. Done. This required playing with self.this and self.thisown to transfer the underlying C++ object but works fine ultimately